python/dynamic-workflow-examples/recovery.py

```diff
@@ -0,0 +1,68 @@
+import ray
+from ray import workflow
+ray.init(namespace="x")
+workflow.init()
+from pathlib import Path
+from filelock import FileLock
+failure_flag = Path("/tmp/failure")
+import time
+
+@workflow.step
+def identity(v):
+    time.sleep(1)
+    if v % 2 == 0:
+        raise RuntimeError()
+    return v
+
+@workflow.step
+def double(v):
+    time.sleep(1)
+    return v * 2
+
+
+@workflow.step
+def s(v):
+    s = 0
+    ss = 0
+    for i in range(v):
+        print(f"--- in iteration {i} ---")
+        try:
+            # this also work
+            # s += workflow.get(identity.step(i))
+            # ss += workflow.get(double.step(i))
+
+            # wait in parallel
+            x = workflow.get([identity.step(i), double.step(i)])
```

Comment:
s_1
s -> [a, b] ->c ===> s -> a -> c
                                    | - > b/


@workflow.step
def f() -> Dict:
    access external service and errored

s_1 -> [id1 -> f(exception), double1] -> s_2 -> [id2, double2] -> s_3
                           |-> (DIct, Optional[Exception])